lib/twingly/search/post.rb

```diff
@@ -46,10 +50,10 @@ def set_values(params)
         @text          = params.fetch('text')
         @language_code = params.fetch('languageCode')
         @location_code = params.fetch('locationCode')
-        @coordinates   = params.fetch('coordinates', {})
-        @links         = params.fetch('links', [])
-        @tags          = params.fetch('tags', [])
-        @images        = params.fetch('images', [])
+        @coordinates   = params.fetch('coordinates') { EMPTY_HASH }
+        @links         = params.fetch('links') { EMPTY_ARRAY }
```

Comment:
Hehe, as I gather doing something like `@links << "http://my.com` would make it pop up in tags and images too. Thinking this might be one optimization step too far, how much gain was it from just going to the block version?

Reply:
My bad, `freeze` was "deep" freeze in this case. Thought it just concerned the allocation of the constant.

Reply:
Yeah, I think we could go with new arrays/hashes here. It will just be one extra object per post and attribute if it's nil. If we use this, we might want to add a specific mutation test.

Reply:
Current way is fine. I thought I would be able to still change the hashes/arrays, but apparently not :) I had some recollection that freeze didn't protect as things like that, but might have changed or I mixed it up with C#'s readonly.

Reply:
I think the problem you're thinking of is the content in an array, that's not explicitly frozen. I.e. `["a"].freeze` would allow the string to be mutated, though no new items can be added. But with empty arrays it should be safe.